src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -35,6 +35,215 @@ public static partial class Base64
         ///   or if the input is incomplete (i.e. not a multiple of 4) and <paramref name="isFinalBlock"/> is <see langword="true"/>.
         /// </returns>
         public static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> bytes, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
+        {
+            // Validation must occur prior to decoding as the actual length will impact future calculations
+            bool containsIgnoredBytes = utf8.IndexOfAny(BytesToIgnore) != -1;
+            if (containsIgnoredBytes)
+            {
+                // Create a new span without bytes to be ignored
+                Span<byte> utf8WithIgnoredBytesRemoved = stackalloc byte[utf8.Length];
```

Comment:
What will happen if `utf8.Length` is huge? Stackoverflow. Stack allocation _must_ be guarded by some reasonable threshold (e.g. 256 byte), otherwise fall back to renting an array from the ArrayPool or if considered as rare-path just allocate (ToArray).